### PR TITLE
[[ Bug 11039 ]] throw error when changing executing behavior

### PR DESCRIPTION
**Pull request by montegoulding**
Now that I noticed there was a partial fix to the issue on develop I have cherry-picked and then added a fix for when setting to empty.

### CODE REVIEW

File: tests/lcs/core/engine/behavior.livecodescript [hunk 1]
```diff
@@ -148,4 +148,26 @@ on TestBehaviorHandlerResolveOrder
...
+end TestBehaviorHandlerResolveOrder
+
+on TestChangeBehaviorWhileExecuting
+   create stack "test"
+   set the defaultStack to "test"
+   create button "child"
+   set the script of button "child" to "on Foo;Bar;end Foo"
+   create button "parent"
+   set the script of button "parent" to "on Bar;Baz;end Bar"
+   set the behavior of button "child" to the long id of button "parent"
```
Comment (livecodeali) [inline tests/lcs/core/engine/behavior.livecodescript]: Is this line meant to be here? The first test seems to be saying the behavior starts off empty

Comment (montegoulding) [reply tests/lcs/core/engine/behavior.livecodescript]: Ah good catch!

File: tests/lcs/core/engine/behavior.livecodescript [hunk 2]
```diff
@@ -148,4 +148,26 @@ on TestBehaviorHandlerResolveOrder
...
+   create stack "test"
+   set the defaultStack to "test"
+   create button "child"
+   set the script of button "child" to "on Foo;Bar;end Foo"
+   create button "parent"
+   set the script of button "parent" to "on Bar;Baz;end Bar"
+   set the script of stack "test" to "on Bar;set the behavior of button " & quote & "child" & quote & " to the long id of button " & quote & "parent"& quote &";end Bar"
+   TestAssertDoesNotThrow "can set behavior while executing if it is empty", "Foo", \
+      the long id of button "child"
+
```
Comment (livecodeali) [inline tests/lcs/core/engine/behavior.livecodescript:163]: Now it looks like there's a missing 
```
set the behavior of button "child" to the long id of button "parent"
```
here

Comment (montegoulding) [reply tests/lcs/core/engine/behavior.livecodescript:163]: The test does it

Comment (livecodeali) [reply tests/lcs/core/engine/behavior.livecodescript:163]: doh!

**Comment by livecodeali** [comment]
@livecode-vulcan review ok 15a880e

**Comment by livecode-vulcan** [comment]
💙 review by @livecodeali ok 15a880ee02692fb9f90bd3ff1260911cf950ea66

**Comment by livecode-vulcan** [comment]
😎 test success 15a880ee02692fb9f90bd3ff1260911cf950ea66

* try-community-armv6-android-api8: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-api8/builds/1029)
* try-community-armv6-android-api9: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-api9/builds/1045)
* try-community-js-emscripten-sdk1.35: [success](https://vulcan.livecode.com/builders/try-community-js-emscripten-sdk1.35/builds/1019)
* try-community-universal-ios-iphoneos11.1: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphoneos11.1/builds/92)
* try-community-universal-ios-iphonesimulator11.1: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphonesimulator11.1/builds/92)
* try-community-universal-mac-macosx10.6: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.6/builds/1024)
* try-community-universal-mac-macosx10.9: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.9/builds/1046)
* try-community-x86-linux-debian7: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian7/builds/1029)
* try-community-x86-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian8/builds/1045)
* try-community-x86_64-linux-debian7: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian7/builds/1028)
* try-community-x86_64-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian8/builds/1038)
* try-community-x86-win32: [success](https://vulcan.livecode.com/builders/try-community-x86-win32/builds/1061)
* try-community-x86_64-win32: [success](https://vulcan.livecode.com/builders/try-community-x86_64-win32/builds/1024)

**Pull request merged**